cni-plugin/pkg/install/install.go

```diff
@@ -401,6 +401,12 @@ func writeCNIConfig(c config) {
 		logrus.Fatal(err)
 	}
 
+	// Safeguarding since WriteFile does not change existing file's permissions.
+	err = os.Chmod(path, perm)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
```

Comment:
weird, this comes from https://github.com/projectcalico/calico/pull/9656, but it's already merged